bip/t3dn_bip/utils.py

```diff
@@ -7,47 +7,39 @@
 from pathlib import Path
 from zlib import decompress
 from array import array
-from .formats import BIP2_MAGIC, PILLOW_FORMATS
+from .formats import test_formats, BIP_FORMATS, PIL_FORMATS, MAGIC_LENGTH
 
 USER_SITE = site.getusersitepackages()
 
 if USER_SITE not in sys.path:
     sys.path.append(USER_SITE)
 
-
 Image = None
-format_supported = {}
 
 
-def _try_import_pillow():
+def _import_pillow():
+    '''Import Pillow and test which formats are supported.'''
     global Image
+
     try:
-        if not Image:
-            from PIL import Image
-            for name, spec in PILLOW_FORMATS.items():
-                supported = True
-                for test in spec.tests:
-                    try:
-                        with Image.open(io.BytesIO(test)) as image:
-                            image.convert('RGBA')
-                    except:
-                        supported = False
-                format_supported[name] = supported
-        return True
+        from PIL import Image
     except ImportError:
-        return False
+        pass
+    else:
+        test_formats()
 
 
-_try_import_pillow()
+_import_pillow()
 
 
 def support_pillow() -> bool:
     '''Check whether Pillow is installed.'''
-    return _try_import_pillow()
+    global Image
```

Comment:
I think we do not need global here, as we are not setting it here and it is defined before.

Reply:
Fair enough.